### PR TITLE
[dcp-756] Add endpoint to query an archive job

### PR DESCRIPTION
**Pull request by ke4**
dcp-756

Changes:
- Add endpoint to retrieve an archive job

### CODE REVIEW

**Comment by prabh-t** [comment]
I am not sure why I am not able to Add a review or Approve, but we have already looked at this together yesterday @ke4 so consider approved.

File: src/main/java/org/humancellatlas/ingest/archiving/submission/web/ArchiveJobController.java [hunk 1]
```diff
@@ -39,13 +40,24 @@ ResponseEntity<?> createArchiveJob(@RequestBody ArchiveJob archiveJob,
...
         archiveJob.setOverallStatus(ArchiveJob.ArchiveJobStatus.PENDING);
     }
 
+    @GetMapping("/archiveJobs/{id}")
+    ResponseEntity<?> getArchiveJob(@PathVariable String id,
+                                    PersistentEntityResourceAssembler resourceAssembler) {
+        Optional<ArchiveJob> archiveJob = archiveJobRepository.findById(id);
+
+        if (archiveJob.isEmpty()) {
+            return ResponseEntity.notFound().build();
```
Comment (amnonkhen) [inline src/main/java/org/humancellatlas/ingest/archiving/submission/web/ArchiveJobController.java:57]: Would it make sense to report the id here?

**Review by amnonkhen — APPROVED** [review]

File: src/main/java/org/humancellatlas/ingest/archiving/submission/web/ArchiveJobController.java (hunk 1 above)
Comment (ke4) [reply src/main/java/org/humancellatlas/ingest/archiving/submission/web/ArchiveJobController.java:57]: @amnonkhen The client already knows the `id`. It seems to me a useless info.

File: src/integration/java/org/humancellatlas/ingest/archiving/web/ArchiveJobControllerTest.java [hunk 2]
```diff
@@ -30,26 +34,101 @@ public class ArchiveJobControllerTest {
...
+        ArchiveJob archiveJob = new ArchiveJob();
+        archiveJob.setId(ARCHIVE_JOB_ID);
+        archiveJob.setCreatedDate(Instant.now());
+        archiveJob.setOverallStatus(status);
+        archiveJob.setSubmissionUuid(submissionUuid);
+
+        return archiveJob;
+    }
+
+    private void setRArchiveResult(ArchiveJob anArchiveJob) {
```
[the remark this answers is not in the document]
Comment (ke4) [reply src/integration/java/org/humancellatlas/ingest/archiving/web/ArchiveJobControllerTest.java]: Thanks. Fixed and pushed.

**Pull request merged**